ufl/algorithms/apply_geometry_lowering.py

```diff
@@ -238,7 +238,7 @@ def circumradius(self, o):
         edges = CellEdgeVectors(domain)
         num_edges = edges.ufl_shape[0]
         j = Index()
-        elen = [sqrt(edges[e, j] * edges[e, j]) for e in range(num_edges)]
+        elen = [sqrt(abs(edges[e, j] * conj(edges[e, j]))) for e in range(num_edges)]
```

Comment:
I think that we should advertise that all of these things come out as `real`. Since they are sometimes later used in a context where that is required (e.g. computing the minimum of the edge lengths below).

So I would prefer that all of these are written as: 

```python
[real(sqrt(real(edges[e, j] * conj(edges[e, j])))) for e in range(num_edges)]
```

etc...